src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets

```diff
@@ -135,10 +136,35 @@
     <Message Importance="normal" Text="Resolved database package references: @(DacpacReference)" />
   </Target>
 
+  <!--
+    Gets a list of included files in the pre- and post-deploy scripts to include in the incremental build Inputs
+  -->
+  <Target Name="GetIncludedFiles" DependsOnTargets="ValidateEnvironment">
+    <!-- Build arguments for the command line tool  -->
+    <PropertyGroup>
+      <PreDeploymentScriptArgument>@(PreDeploy->'--predeploy %(Identity)', ' ')</PreDeploymentScriptArgument>
+      <PostDeploymentScriptArgument>@(PostDeploy->'--postdeploy %(Identity)', ' ')</PostDeploymentScriptArgument>
+      <DebugArgument Condition="'$(MSBuildSdkSqlProjDebug)' == 'True'">--debug</DebugArgument>
+      <DacpacToolCommand>dotnet &quot;$(DacpacToolExe)&quot; collect-includes $(PreDeploymentScriptArgument) $(PostDeploymentScriptArgument) $(DebugArgument)</DacpacToolCommand>
+    </PropertyGroup>
+    <!-- Run it, except during design-time builds -->
+    <Message Importance="low" Text="Running command: $(DacpacToolCommand)" />
+    <Exec Command="$(DacpacToolCommand)" ConsoleToMsBuild="true" EchoOff="true" StandardOutputImportance="Low"
+          Condition="'$(DesignTimeBuild)' != 'true' AND '$(BuildingProject)' == 'true'">
+      <Output TaskParameter="ConsoleOutput" PropertyName="IncludedFiles" />
```

Comment:
Isn't there an `ItemName` attribute here as well that we can use, rather than having to convert it into an ItemGroup manually later on? Looking at the [docs for the Exec task](https://docs.microsoft.com/en-us/visualstudio/msbuild/exec-task?view=vs-2019#parameters) it seems to be an `ITaskItem[]` property.

Reply:
Confirmed I can just use `$(IncludedFiles)` instead of converting to an ItemGroup, I didn't realize that Properties could be iterated over like ItemGroups, but it seems to be working to me.